lib/src/err/mod.rs

```diff
@@ -770,6 +772,55 @@ pub enum Error {
 	/// The db is running without an available storage engine
 	#[error("The db is running without an available storage engine")]
 	MissingStorageEngine,
+
+	/// Error that can happen at any point during bootstrap
+	/// TODO change this to heartbeat? cleanup? cluster lifecycle?
+	/// there is overlap between bootstrap and heartbeat, but then there will be errors in lq that
+	/// arent bootstrap (definitely) or heartbeat (maybe)
+	#[error("Bootstrap failure: {0}")]
+	BootstrapError(BootstrapCause),
+}
+
+#[derive(Error, Debug)]
+#[non_exhaustive]
+pub enum BootstrapCause {
+	#[error("Failed to send to channel: {0}")]
+	ChannelSendError(ChannelVariant),
+	#[error("Failed to recv from channel: {0}")]
+	ChannelRecvError(ChannelVariant),
```

Comment:
These happen when the other end of the channel is closed or dropped. This is not something we can expect a user to handle. Can't we handle these internally and avoid returning an error?

Reply:
Yes, it is because of closed channel, but it can be because we requested a transaction but no transaction is received. Agree users can't handle it, but it is good for testing - we can then close channels in tests and validate how that gets handled and what the end state is under such handling. Hypothetical argument.

Maybe we should have a separation of internal vs external errors? So internally, we do what we can for clear execution. And externally we translate internal errors to only those that can be handled? Ties with the panics, as you mentioned we lose the context. We would lose some context in that model.

Reply:
Removed, as only usage was in fact around send failed and receive failed - they are now panicking